packages/service-library/src/servicelib/volumes_utils.py

```diff
@@ -0,0 +1,54 @@
+from pathlib import Path
+from typing import Final, Optional
+
+import aiofiles
+from pydantic import BaseModel, Field, root_validator
+
+HIDDEN_FILE_NAME: Final[str] = ".hidden_do_not_remove"
+AGENT_FILE_NAME: Final[str] = ".agent"
+
+
+class VolumeState(BaseModel):
+    requires_saving: bool = Field(
+        ...,
+        description=(
+            "when True, the contents of the volume must be saved whe "
+            "closing the service"
+        ),
+    )
+    was_saved: Optional[bool] = Field(
```

Comment:
OK I understand it now: ``was_saved`` is complementing  ``requires_saving=True``.   I still find it unintuitive.  IMO the problem is the design of the model itself. It reminds me a discussion in  "Product and Sum Types" (pg. 53) of Viafore's book "Robust Python" ( I think we even already discussed about this in the past). I think you should have a look at it for more insight on how to design settings in a scenario like this.

If you do not want to change the design, I would rather go for **``was_saved=False`` as default value** instead of ``None`` since if saving is NOT required then ``was_saved=False`` also makes sense. And if saving IS required, the initial state of ``was_saved`` I guess still has to be ``False`` until the user changes it.  Nonetheless, even in this case it feels strange because, as I said above, I think the design is faulty.

Reply:
I took the time to simplify the design. Let me know if makes more sense now.